index.ts

```diff
@@ -593,102 +688,74 @@ commander
-                  if (path.extname(name) === '.gs') { // only print error once (for .gs)
-                    logError(null, ERROR.CONFLICTING_FILE_EXTENSION(fileNameWithoutExt));
-                  }
-                } else if (path.extname(name) === '.gs') {
-                  // rename file to js
-                  console.log(LOG.RENAME_FILE(fileNameWithoutExt + '.gs', fileNameWithoutExt + '.js'));
-                  fs.renameSync(fileNameWithoutExt + '.gs', fileNameWithoutExt + '.js');
-                }
-              });
-              if (abortPush) return spinner.stop(true);
-
-              const files = filePaths.map((name, i) => {
-                let nameWithoutExt = name.slice(0, -path.extname(name).length);
-                // Replace OS specific path separator to common '/' char
-                nameWithoutExt = nameWithoutExt.replace(/\\/g, '/');
-
-                // Formats rootDir/appsscript.json to appsscript.json.
-                // Preserves subdirectory names in rootDir
-                // (rootDir/foo/Code.js becomes foo/Code.js)
-                let formattedName = nameWithoutExt;
-                if (rootDir) {
-                  formattedName = nameWithoutExt.slice(
-                    rootDir.length + 1,
-                    nameWithoutExt.length
-                  );
-                }
-                if (getAPIFileType(name) && !anymatch(ignorePatterns, name)) {
-                  nonIgnoredFilePaths.push(name);
-                  const file: AppsScriptFile = {
-                    name: formattedName, // the file base name
-                    type: getAPIFileType(name), // the file extension
-                    source: contents[i] //the file contents
-                  };
-                  return file;
-                } else {
-                  return; // Skip ignored files
-                }
-              }).filter(Boolean); // remove null values
-
-              script.projects.updateContent({
-                scriptId,
-                resource: { files }
-              }, {}, (error: any, res: Function) => {
-                spinner.stop(true);
-                if (error) {
-                  console.error(LOG.PUSH_FAILURE);
-                  error.errors.map((err: any) => {
-                    console.error(err.message);
-                  });
-                  console.error(LOG.FILES_TO_PUSH);
-                  nonIgnoredFilePaths.map((filePath) => {
-                    console.error(`└─ ${filePath}`);
-                  });
-                } else {
-                  nonIgnoredFilePaths.map((filePath) => {
-                    console.log(`└─ ${filePath}`);
-                  });
-                  console.log(LOG.PUSH_SUCCESS(nonIgnoredFilePaths.length));
-                }
-              });
-            });
+        getProjectFiles(rootDir, (err,projectFiles,files) => {
+          if (err != undefined) {
+            return;
+          } else if(projectFiles!=undefined) {
```

Comment:
Format the PR here and elsewhere
`} else if (!projectFiles) {`

Reply:
I tried `!projectFiles` but it gave me an error saying it could still be undefined.